GomokuClient/packages/gomoku-core/src/hooks/useGameSession.ts

```diff
@@ -5,13 +5,14 @@ import { useSignalRConnection } from "@/context";
 import { SwaggerServices, SwaggerTypes } from "@/api";
 import { useQuery } from "@tanstack/react-query";
 
-export const useGameSession = (gameID: string) => {
+export const useGameSession = (gameID?: string) => {
   const { tiles, setTiles, winner, addTile, lastTile, setLastTile } =
     useTiles();
   const { hubProxy, isConnected, registerEventHandlers } =
     useSignalRConnection();
 
-  const { data: gameHistory } = useGameHistory(gameID);
+  // TODO: check server acceptance of empty string errors
+  const { data: gameHistory } = useGameHistory(gameID || "");
```

Comment:
Maybe throw error if hook called without gameId?

Reply:
Then we have to put a lot of logic in query hooks and I try to avoid that as it increases complexity by a lot

<img width="1176" alt="Screenshot 2024-10-13 at 17 07 09" src="https://github.com/user-attachments/assets/6800b4c4-d57e-4bd8-a771-14a0efa47c5c">